### PR TITLE
feat(pub-techdocs): make checking out `submodules` configurable

### PR DESCRIPTION
**Pull request by zzehring**
This allows the user to pass in different options to `actions/checkout` submodules option. Default is `false`.

### CODE REVIEW

File: .github/workflows/publish-techdocs.md [hunk 1]
```diff
@@ -32,11 +32,11 @@ jobs:
...
-| `checkout-repo`             | boolean | Should this workflow also check out the current repository? (default: `true`) |
-| `publish`                   | boolean | Enable disable publishing the built docs to EngHub (default: `true`)          |
+| Name                        | Type    | Description                                                                             |
+| --------------------------- | ------- | --------------------------------------------------------------------------------------- |
+| `namespace`                 | string  | The entity's namespace within EngHub (usually `default`)                                |
+| `kind`                      | string  | The kind of the entity in EngHub (usually `component`)                                  |
+| `name`                      | string  | The name of the entity in EngHub (usually matches the name of the repository)           |
+| `default-working-directory` | string  | The directory where the techdocs-cli should be run (default: `.`)                       |
+| `publish`                   | boolean | Enable disable publishing the built docs to EngHub (default: `true`)                    |
```
Comment (iainlane) [inline .github/workflows/publish-techdocs.md:41]:
```suggestion
| `publish`                   | boolean | Whether to publish the built docs to EngHub (default: `true`)                    |
```

(sorry that's probably going to mess up the alignment of the table)

**Review by iainlane — APPROVED** [review]
did this cause a problem? be good to say so in the commit message :slightly_smiling_face:

File: .github/workflows/publish-techdocs.md (hunk 1 above)
Comment (zzehring) [reply .github/workflows/publish-techdocs.md:41]: Thank goodness for extensions 🙏

**Comment by zzehring** [comment]
> did this cause a problem? be good to say so in the commit message 🙂

No issues. Just wanted to make this option to be a bit more complete.

**Pull request merged**